software/profiling/Vampir.rst

```diff
@@ -1,11 +1,320 @@
+ | Launching VampirServer...
+ | Submitting LSF batch job (this might take a while)...
+ | Warning: more than 1 task/rank assigned to a core
+ | VampirServer 9.11.1 OLCF (4626dba5)
+ | Licensed to ORNL
+ | Running 4 analysis processes... (abort with vampirserver stop 10102)
+ | User: Test
+ | Password: XXXXXXXXXXXX
+ | VampirServer <10102> listens on: h50n05:30040
+
+----------------------------------------------------------------------
+
+
+Launch the Vampir GUI
+---------------------
+.. code::
+
+   $ vampir &
+
+.. _vampauth:
+
+Once the GUI has opened, you will need to connect to the Vampirserver using the
+``Remote File`` option as shown below. **If there is a 'recent files' window open, select 'open other'.**
+Enter the ``node ID`` and the ``port`` number and press 'Connect'.
+
+
+.. image:: /images/vampir_open_remote_location.png
+   :align: left
+   :width: 800px
+   :height: 300px
+
+|
+|
+|
+|
+|
+|
+|
+|
+|
+|
+|
+|
+|
+
+When the server authentication window pops up, you will need to enter your ``USERID``
+& the :ref:`Vampirserver password <vampserpw>` that was printed on the terminal screen. Once authenticated, you will be able to navigate through the filesystem to your **.otf2** files
+
+.. image:: /images/vampir_server_auth.png
+   :align: left
+   :width: 400px
+   :height: 300px
+
+
+|
+|
+|
+|
+|
+|
+|
+|
+|
+|
+|
+|
+|
+|
+
+Vampir Tunneling to Vampirserver
+===========================================
+
+.. image:: /images/vampir_reverse_connect_gui_to_compute.png
+   :align: left
+   :width: 350px
+   :height: 125px
+
+|
+|
+|
+|
+|
+|
+
+.. Tip::
+
+   This connection method is more complex than the other 2 methods, however it also can provide a more optimal experience for very large trace files.
+
+
+
+.. attention::
+
+   ``This method will require you to have a local copy of the Vampir GUI already installed on your machine.``
+
+   If you do not have a local copy, please reach out to the help desk at help@olcf.ornl.gov for instructions on getting a local copy.
+
+Similar to the previous methods outlined above, you will start by connecting to `Summit <https://docs.olcf.ornl.gov/systems/summit_user_guide.html#connecting>`_.
+Once connected you will then need to start the :ref:`Vampirserver. <vamps>`
+
+.. code::
+
+   $ module load vampir
+
+   #Start the VampirServer
+
+   $ vampirserver start -- -P <projectID> -w <walltime> -q <queue>
+
+After the Vampirserver is started successfully, you will need the information printed on the :ref:`terminal window. <vampserpw>`
+That includes:
+
+* Node ID
+* Port Number
+* password
+
+
+Once the Vampirsever is started, in a **fresh** terminal window on your **Local** machine you can then initiate the
+port forward command. This will open a secure tunnel from your local machine to the backend server running Vampirserver.
+
+**Port Forwarding**
+
+.. code::
+
+   SSH -L <localport>:<Node ID>:<Remote port>  <USERID>$summit.olcf.ornl.gov
```

Comment:
Here SSH should be ssh in case anyone copies and pastes.
Also, I didn't actually test this method since I don't have a local copy of Vampir on my laptop. Is <Node ID> what should be used here? I don't actually know, so if I'm wrong, just ignore me on this :)